GrandCentralBoard/Widgets/Bonus/Scene/BonusScene.swift

```diff
@@ -37,9 +37,23 @@ class BonusScene: SKScene {
             world.addChild(bubble)
         }
     }
+
+    override func update(currentTime: NSTimeInterval) {
+        guard !world.children.isEmpty else { return }
+
+        let nodesFitScreen = CGRectContainsRect(frame, world.calculateAccumulatedFrame())
+        if !nodesFitScreen {
+            scaleBubblesDown()
+        }
+    }
     
     private func updateWithViewModel(viewModel: BonusWidgetViewModel) {
         self.viewModel = viewModel
+        addOrUpdateBubbles()
+        removeOldBubbleIfNecessary()
```

Comment:
I don't agree. I want to have this logic stated explicity in `updateWithViewModel` method.